<a href="https://colab.research.google.com/github/cristiangodoyangel/Cristian_Godoy_Examen_2_intento_2/blob/main/Cristian_Godoy_Examen_2_intento_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

CCC

# Clasificación con el Bank Marketing Dataset

## Objetivo

Este examen tiene como objetivo evaluar la capacidad del estudiante para limpiar, explorar, implementar y evaluar modelos de clasificación en un dataset de clientes bancarios. Utilizaremos el “Bank Marketing Dataset” disponible en UCI Machine Learning Repository. Los estudiantes deben demostrar habilidades prácticas en la manipulación de datos, creación de visualizaciones y modelado predictivo.

## Descripción del Dataset

El dataset **Bank Marketing Dataset**, contiene información sobre campañas de marketing directo (llamadas telefónicas) de una institución bancaria portuguesa.

El objetivo es predecir si un cliente suscribirá un depósito a plazo fijo (variable objetivo `y`), a partir de datos personales y del historial de contacto previo.


In [7]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# se carga dataset de vinos blancos
path = "/content/drive/MyDrive/0002 Python - Data Science/Datasets/Examen/02/bank-full.csv"
df = pd.read_csv(path, sep=";")

In [6]:
df.head()

age           job  marital  education default  balance housing loan  \
0   58    management  married   tertiary      no     2143     yes   no   
1   44    technician   single  secondary      no       29     yes   no   
2   33  entrepreneur  married  secondary      no        2     yes  yes   
3   47   blue-collar  married    unknown      no     1506     yes   no   
4   33       unknown   single    unknown      no        1      no   no   

   contact  day month  duration  campaign  pdays  previous poutcome   y  
0  unknown    5   may       261         1     -1         0  unknown  no  
1  unknown    5   may       151         1     -1         0  unknown  no  
2  unknown    5   may        76         1     -1         0  unknown  no  
3  unknown    5   may        92         1     -1         0  unknown  no  
4  unknown    5   may       198         1     -1         0  unknown  no

## Entendiendo el Dataset

Antes de cualquier avance, es importante conocer  el dataset. así entiendo lo difícil del problema, entendiendo el target (lo que buscamos) sabré si debo escalar o codificar.
 Resivo el tipo de columnas.


In [13]:
# Se revisa el df para entenderlo
print("\n")
print("========================================================================================================================================")
print("Dimensiones del dataset:", df.shape)
print("========================================================================================================================================")
print("\n")
df.head()



Dimensiones del dataset: (45211, 17)




age           job  marital  education default  balance housing loan  \
0   58    management  married   tertiary      no     2143     yes   no   
1   44    technician   single  secondary      no       29     yes   no   
2   33  entrepreneur  married  secondary      no        2     yes  yes   
3   47   blue-collar  married    unknown      no     1506     yes   no   
4   33       unknown   single    unknown      no        1      no   no   

   contact  day month  duration  campaign  pdays  previous poutcome   y  
0  unknown    5   may       261         1     -1         0  unknown  no  
1  unknown    5   may       151         1     -1         0  unknown  no  
2  unknown    5   may        76         1     -1         0  unknown  no  
3  unknown    5   may        92         1     -1         0  unknown  no  
4  unknown    5   may       198         1     -1         0  unknown  no

In [14]:
# se revisan los tipos de columnas
print("\n")
print("========================================================================================================================================")
print("Dimensiones del dataset:", df.shape)
print("========================================================================================================================================")
print("\n")
df.info()



Dimensiones del dataset: (45211, 17)


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45211 entries, 0 to 45210
Data columns (total 17 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   age        45211 non-null  int64 
 1   job        45211 non-null  object
 2   marital    45211 non-null  object
 3   education  45211 non-null  object
 4   default    45211 non-null  object
 5   balance    45211 non-null  int64 
 6   housing    45211 non-null  object
 7   loan       45211 non-null  object
 8   contact    45211 non-null  object
 9   day        45211 non-null  int64 
 10  month      45211 non-null  object
 11  duration   45211 non-null  int64 
 12  campaign   45211 non-null  int64 
 13  pdays      45211 non-null  int64 
 14  previous   45211 non-null  int64 
 15  poutcome   45211 non-null  object
 16  y          45211 non-null  object
dtypes: int64(7), object(10)
memory usage: 5.9+ MB


## Manejo de nulos o inválidos

Al parecer no hay información vacía, lo vemos en 45211 non-null, pero al revisar con .head() vimos que hay "unknow" en las columnas postcome y contact, revisaré si son necesarias para el target, si no es así se eliminarán.